### PR TITLE
Add --git-archive-path and --metadata-archive-path options

### PR DESCRIPTION
**Pull request by jfine**
Adds the ability to migrate a local file utilizing the --git-archive-path and --metadata-archive-path options. This change is similar to the --git-archive-url and --metadata-archive-url options though instead of starting the migration directly it uploads the archives (only one upload if the same archive is specified) to the storage option selected and then starts the migration.

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [ ] Issue linked
- [ ] Docs updated (or issue created)
- [ ] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

### CODE REVIEW

File: src/gei/Commands/MigrateRepo/MigrateRepoCommandArgs.cs [hunk 1]
```diff
@@ -25,6 +25,8 @@ public class MigrateRepoCommandArgs : CommandArgs
         public bool NoSslVerify { get; set; }
         public string GitArchiveUrl { get; set; }
         public string MetadataArchiveUrl { get; set; }
+        public string GitArchiveFilePath { get; set; }
+        public string MetadataArchiveFilePath { get; set; }
```
Comment (ArinGhazarian) [inline src/gei/Commands/MigrateRepo/MigrateRepoCommandArgs.cs:29]: I just check the `bbs2gh migrate-repo` command and noticed that we used `ArchivePath` so for consistency it's better to remove the file form the variable names:
```suggestion
        public string GitArchivePath { get; set; }
        public string MetadataArchivePath { get; set; }
```

File: src/gei/Commands/MigrateRepo/MigrateRepoCommandArgs.cs [hunk 2]
```diff
@@ -41,11 +43,26 @@ public override void Validate(OctoLogger log)
...
+            {
+                throw new OctoshiftCliException("The options --metadata-archive-url and --metadata-archive-file-path may not be used together");
+            }
+
             if (string.IsNullOrWhiteSpace(GitArchiveUrl) != string.IsNullOrWhiteSpace(MetadataArchiveUrl))
             {
                 throw new OctoshiftCliException("When using archive urls, you must provide both --git-archive-url --metadata-archive-url");
             }
 
+            if (string.IsNullOrWhiteSpace(GitArchiveFilePath) != string.IsNullOrWhiteSpace(MetadataArchiveFilePath))
```
Comment (ArinGhazarian) [inline src/gei/Commands/MigrateRepo/MigrateRepoCommandArgs.cs:61]:
```suggestion
            if (GitArchiveFilePath.IsNullOrWhiteSpace() != MetadataArchiveFilePath.IsNullOrWhiteSpace())
```

File: src/gei/Commands/MigrateRepo/MigrateRepoCommandArgs.cs [hunk 3]
```diff
@@ -41,11 +43,26 @@ public override void Validate(OctoLogger log)
...
 
+            if (GitArchiveUrl.HasValue() && GitArchiveFilePath.HasValue())
+            {
+                throw new OctoshiftCliException("The options --git-archive-url and --git-archive-file-path may not be used together");
+            }
+
+            if (MetadataArchiveUrl.HasValue() && MetadataArchiveFilePath.HasValue())
+            {
+                throw new OctoshiftCliException("The options --metadata-archive-url and --metadata-archive-file-path may not be used together");
+            }
```
Comment (ArinGhazarian) [inline src/gei/Commands/MigrateRepo/MigrateRepoCommandArgs.cs:54]: Are we going to support a scenario where `GitArchiveUrl` is provided with `MetadataArchivePath` or vice versa? if not then we need to validate those scenarios too.

File: src/gei/Commands/MigrateRepo/MigrateRepoCommand.cs [hunk 4]
```diff
@@ -120,6 +118,16 @@ public MigrateRepoCommand() : base(
             IsHidden = true,
             Description = "An authenticated SAS URL to an Azure Blob Storage container with a pre-generated metadata archive. Only used when an archive has been generated and uploaded prior to running a migration (not common). Must be passed in when also using --git-archive-url"
         };
+        public Option<string> GitArchiveFilePath { get; } = new("--git-archive-file-path")
```
Comment (ArinGhazarian) [inline src/gei/Commands/MigrateRepo/MigrateRepoCommand.cs:121]:
```suggestion
        public Option<string> GitArchivePath { get; } = new("--git-archive-path")
```

File: src/gei/Commands/MigrateRepo/MigrateRepoCommand.cs [hunk 5]
```diff
@@ -120,6 +118,16 @@ public MigrateRepoCommand() : base(
...
+        {
+            IsHidden = true,
+            Description = "Used to migrate an archive that is on disk, must be used with --metadata-archive-file-path"
+        };
+        public Option<string> MetadataArchiveFilePath { get; } = new("--metadata-archive-file-path")
```
Comment (ArinGhazarian) [inline src/gei/Commands/MigrateRepo/MigrateRepoCommand.cs:126]:
```suggestion

        public Option<string> MetadataArchivePath { get; } = new("--metadata-archive-path")
```

File: src/gei/Commands/MigrateRepo/MigrateRepoCommandArgs.cs (hunk 3 above)
Comment (jfine) [reply src/gei/Commands/MigrateRepo/MigrateRepoCommandArgs.cs:54]: Probably not but the combination of MetadataArchiveUrl requiring GitArchiveUrl but also not allowing the path settings currently prevent it.

File: src/gei/Commands/MigrateRepo/MigrateRepoCommand.cs (hunk 5 above)
Comment (jfine) [reply src/gei/Commands/MigrateRepo/MigrateRepoCommand.cs]: Updated via https://github.com/github/gh-gei/pull/1294/commits/1f328da4b48687841cfb9edf4a557d83454f0c74

File: src/gei/Commands/MigrateRepo/MigrateRepoCommand.cs (hunk 4 above)
Comment (jfine) [reply src/gei/Commands/MigrateRepo/MigrateRepoCommand.cs]: Updated via https://github.com/github/gh-gei/pull/1294/commits/1f328da4b48687841cfb9edf4a557d83454f0c74

File: src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs [hunk 6]
```diff
@@ -287,7 +292,34 @@ private string ExtractGhesBaseUrl(string ghesApiUrl)
...
+#pragma warning disable IDE0046
+        else if (_awsApi.HasValue())
+#pragma warning restore IDE0046
+        {
+            return await UploadArchiveToAws(awsBucketName, archiveUploadFileName, archiveContent);
+        }
+        else
+        {
+            return await UploadArchiveToAzure(archiveUploadFileName, archiveContent);
+        }
```
Comment (ArinGhazarian) [inline src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs:319]: now that the upload method is uploading just one file I think we can switch to a `using declaration` syntax:
```suggestion
        await using var archiveContent = _fileSystemProvider.OpenRead(archiveDownloadFilePath);

        if (useGithubStorage)
        {
            return await UploadArchiveToGithub(githubTargetOrg, archiveUploadFileName, archiveContent);
        }

        if (_awsApi.HasValue())
        {
            return await UploadArchiveToAws(awsBucketName, archiveUploadFileName, archiveContent);
        }

        return await UploadArchiveToAzure(archiveUploadFileName, archiveContent);
```

**Review by ArinGhazarian — COMMENTED** [review]
Changes look good 🎉  The only missing thins is updating the [releasenotes.md](https://github.com/github/gh-gei/blob/main/RELEASENOTES.md)

**Review by ArinGhazarian — APPROVED** [review]

**Pull request merged**